services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java

```diff
@@ -191,6 +194,9 @@ public LoginProtocol create(KeycloakSession session) {
 
         model = UserRealmRoleMappingMapper.create(null, GROUPS, GROUPS, true, true, true);
         builtins.put(GROUPS, model);
+
+        model = AcrProtocolMapper.create(ACR, true, true);
```

Comment:
Good point.

Currently the protocol mapper is hidden in the UI when the feature is disabled. However the client scope cannot be easily hidden (assuming that it was already created) as there is no way to conditionally hide any client scope when some feature is enabled/disabled and I don't think it is possible to easily do it without any hacks...

What I can do is, that when new realm is going to be created (or during migration) and "acr" feature is disabled, then the "acr" client scope (including protocol mapper) won't be created. This means that when administrator later decides to enable the feature, he will need to create client scope (and protocol mapper) manually. This is something, which will need to be documented (I can change the docs PR for this as well).

However the opposite case (when feature is first enabled and then disabled) still means that "acr" client scope will be created and visible. So for example scenario like this:
- Feature is enabled
- New realm is created. This will create "acr" client scope with protocol mapper
- Administrator decides to disable the feature and restart server
- Now the "acr" client scope will still be visible to administrator. But protocol mapper is not visible to the administrator (due the provider is disabled). So administrator will need to delete the "acr" client scope if he wants this completely removed.

I am slightly towards to change it in a way that at the time of the creation, the "acr" client scope will be created based on whether the feature is enabled or disabled at the time of the creation. WDYT?

Reply:
The updated version seems really promising. Thanks @mposolda